src/field/field.cxx

```diff
@@ -32,14 +32,58 @@
 #include <utils.hxx>
 #include <bout/mesh.hxx>
 
-Field::Field(Mesh *localmesh)
-  : fieldmesh(localmesh==nullptr ? bout::globals::mesh : localmesh) {
-
-// Note we would like to do `fieldCoordinates = getCoordinates();` here but can't
-// currently as this would lead to circular/recursive behaviour (getCoordinates would
-// call fieldmesh->coordinates, which would create fields, which would then call
-// getCoordinates again etc.). This also requires care in the derived class
-// constructors.
+Field::Field(Mesh *localmesh, CELL_LOC location_in,
+             DIRECTION xDirectionType_in, DIRECTION yDirectionType_in,
+             DIRECTION zDirectionType_in)
+    : fieldmesh(localmesh==nullptr ? bout::globals::mesh : localmesh),
+      location(location_in), xDirectionType(xDirectionType_in),
+      yDirectionType(yDirectionType_in), zDirectionType(zDirectionType_in) {
+
+  // Need to check for nullptr again, because the fieldmesh might still be
+  // nullptr if the global mesh hasn't been initialized yet
+  if (fieldmesh != nullptr) {
+    // get Coordinates for our location from fieldmesh
```

Comment:
I think it needed a better comment, like `// sets fieldCoordinates by getting Coordinates for our location from fieldmesh`. Since `getCoordinates()` is for getting `fieldCoordinates` *from* the field, this is (to me) a non-obvious use, in that the fact that `getCoordinates()` *sets* `fieldCoordinates` is a sort of side-effect. I'll update the comment.